### PR TITLE
Update to latest api

### PR DESCRIPTION
**Pull request by derekwbrown**
### What does this PR do?

A brief description of the change being made with this pull request.

### Motivation

What inspired you to submit this pull request?

### Additional Notes

Anything else we should know when reviewing?

### CODE REVIEW

File: pkg/ebpf/driver_interface.go [hunk 1]
```diff
@@ -142,18 +153,27 @@ func (dc *DriverInterface) prepareDriverFilters() error {
...
+	filters = append(filters, newDDAPIFilter(C.DIRECTION_OUTBOUND, C.FILTER_LAYER_ALE_CONNECT, iface.Index, true))
+	filters = append(filters, newDDAPIFilter(C.DIRECTION_OUTBOUND, C.FILTER_LAYER_TRANSPORT, iface.Index, true))
+
+	filters = append(filters, newDDAPIFilter(C.DIRECTION_INBOUND, C.FILTER_LAYER_ALE_RECVCONN, iface.Index, true))
+	filters = append(filters, newDDAPIFilter(C.DIRECTION_INBOUND, C.FILTER_LAYER_TRANSPORT, iface.Index, true))
+
+	// this one doesn't really have a "direction".  TODO  Should probably clarify the API
+	filters = append(filters, newDDAPIFilter(C.DIRECTION_INBOUND, C.FILTER_LAYER_ALE_CLOSURE, iface.Index, true))
+
+	// for now skip ipv6
```
Comment (stb2132) [inline pkg/ebpf/driver_interface.go:165]: Is there a reason we're skipping ipv6 for now?

File: pkg/ebpf/driver_interface.go [hunk 2]
```diff
@@ -142,18 +153,27 @@ func (dc *DriverInterface) prepareDriverFilters() error {
...
-	filters = append(filters, newDDAPIFilter(C.DIRECTION_OUTBOUND, iface.Index, true))
-	filters = append(filters, newDDAPIFilter(C.DIRECTION_INBOUND, iface.Index, false))
-	filters = append(filters, newDDAPIFilter(C.DIRECTION_OUTBOUND, iface.Index, false))
+	filters = append(filters, newDDAPIFilter(C.DIRECTION_OUTBOUND, C.FILTER_LAYER_ALE_CONNECT, iface.Index, true))
+	filters = append(filters, newDDAPIFilter(C.DIRECTION_OUTBOUND, C.FILTER_LAYER_TRANSPORT, iface.Index, true))
+
+	filters = append(filters, newDDAPIFilter(C.DIRECTION_INBOUND, C.FILTER_LAYER_ALE_RECVCONN, iface.Index, true))
+	filters = append(filters, newDDAPIFilter(C.DIRECTION_INBOUND, C.FILTER_LAYER_TRANSPORT, iface.Index, true))
+
+	// this one doesn't really have a "direction".  TODO  Should probably clarify the API
```
Comment (stb2132) [inline pkg/ebpf/driver_interface.go:162]: Would you mind letting me know what you mean by doesn't have direction?

**Review by stb2132 — APPROVED** [review]

File: pkg/ebpf/driver_interface.go (hunk 1 above)
Comment (derekwbrown) [reply pkg/ebpf/driver_interface.go:165]: I was being lazy :)

File: pkg/ebpf/driver_interface.go (hunk 2 above)
Comment (derekwbrown) [reply pkg/ebpf/driver_interface.go:162]: the "close" callout doesn't have a direction b/c it's not actually associated with a packet.  It just means the associated file handle has been closed (and, I think you can still get packets afterward).  But it's how we know, for example, that a UDP flow is really gone vs. just idle for a long time.  LMK if that's not clear.

Comment (stb2132) [reply pkg/ebpf/driver_interface.go:162]: Makes sense!

**Pull request merged**